src/model/treewalker.js

```diff
@@ -153,6 +153,34 @@ export default class TreeWalker {
 	}
 
 	/**
+	 * Move {@link #position} in the {@link #direction} skipping values as long as the callback function returns `true`.
+	 *
+	 * For example:
+	 *
+	 * 		walker.skip( value => value.type == 'text' ); // <paragraph>[]foo</paragraph> -> <paragraph>foo[]</paragraph>
+	 * 		walker.skip( value => true ); // Move the position to the end: <paragraph>[]foo</paragraph> -> <paragraph>foo</paragraph>[]
+	 * 		walker.skip( value => false ); // Do not move the position.
+	 *
+	 * @param {Function} skip Callback function. Gets {@link module:engine/model/treewalker~TreeWalkerValue} and should
+	 * return `true` if the value should be skipped or `false` if not.
+	 */
+	skip( skip ) {
```

Comment:
Yea. Good luck ;) This was my first idea but then I realized how complex it would be. This way you can call `skip` multiple times, if you need, with different parameters and the code is simple. Adding more options to the `TreeWalker` we will make to code unmaintainable (in fact it's on the edge now).